mock/mock.go

```diff
@@ -812,7 +812,16 @@ func (args Arguments) Diff(objects []interface{}) (string, int) {
 		}
 
 		if matcher, ok := expected.(argumentMatcher); ok {
-			if matcher.Matches(actual) {
+			var matches bool
+			func() {
+				defer func() {
```

Comment:
Could you remind me why we have to wrap the deferred function call in another function? Is it because of line 822?

Reply:
I chose to do that so that only a panic from the argument matcher function will be caught by this recover. By not having the outer func() any other panics in the testify codebase would be a false positive for a panic in the argument matcher.